lib/src/message_parser.dart

```diff
@@ -0,0 +1,361 @@
+// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Contains a parser for ICU format plural/gender/select format for localized
+/// messages. See extract_to_arb.dart and make_hardcoded_translation.dart.
+library message_parser;
+
+// ignore: import_of_legacy_library_into_null_safe
```

Comment:
This works; you could also exclude this file from null safety (and look to migrate it as part of later work). You can see the exclusion syntax at the start of other files (`// @dart = ...`).